exercises/takudzwa_makoni/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,163 @@
+
+//#include "pch.h"
+#include <iostream>
+#include <stdio.h>
+#include <math.h>
+#include <vector>
+#include <string>
+
+const double pi = (double)3.14159; // constant Pi casted to double
+using namespace std;
+
+
+
+//------------------------- shape class -----------------------------------
+
+class shape {
+
+public:
+	virtual void getNumOfSides()	{ cout << "override" << endl; }
+	virtual void getName()			{ cout << "override" << endl; }
+
+	virtual void getArea()			{ cout << "override" << endl; }
+	virtual void getPerimeter()		{ cout << "override" << endl; }
+};
+
+
+//--------------------------------------------------------------------------
+
+//------------------------- shape subclasses -------------------------------
+
+class square : public shape {
+
+private:
+
+	double sideLength;
+	string name;
+	int numOfSides;
+
+public:
+
+	virtual void getNumOfSides() { cout << numOfSides << endl; }
+	virtual void getName() { cout << name << endl; }
+
+	virtual void getArea() { cout << sideLength * sideLength << endl; }
+	virtual void getPerimeter() { cout << 4 * sideLength << endl; }
+
+	//contructor
+	square::square() {
+		sideLength = 1;
+		name = "square";
+		numOfSides = 4;
+	};
+};
+
+class rectangle : public shape {
+
+private:
+
+	double sideWidth, sideHeight;
+	string name;
+	int numOfSides;
+
+public:
+
+	virtual void getNumOfSides() { cout << numOfSides << endl; }
+	virtual void getName() { cout << name << endl; }
+
+	virtual void getArea()		{ cout << sideHeight * sideWidth << endl; }
+	virtual void getPerimeter()	{ cout << (2 * sideWidth) + (2 * sideHeight) << endl; }
+
+	rectangle::rectangle() {
+		sideWidth = 1;
+		sideHeight = 2;
+		name = "rectangle";
+		numOfSides = 4;
+	};
+};
+
+class isoTriangle : public shape { // isosceles triangle class
+
+private:
+
+	double base, height;
+	string name;
+	int numOfSides;
+
+
+public:
+
+	virtual void getNumOfSides()	{ cout << numOfSides << endl; }
+	virtual void getName()			{ cout << name << endl; }
+
+	virtual void getArea()		{ cout << 0.5 * base* height << endl; }
+	virtual void getPerimeter() { cout << 2 * (sqrt(height * height) + ((base * base) / 4)) << endl; }
+
+	//constructor
+	isoTriangle::isoTriangle() {
+		base	= 1;
+		height	= 2;
```

Comment:
How did you decide that the `base` and `height` were 1 and 2 respectively - if I were to use your `isoTriangle` how would I be able to provide `base`, `height`?